eclair-core/src/main/scala/fr/acinq/eclair/crypto/ChannelKeyManager.scala

```diff
@@ -109,23 +106,10 @@ trait KeyManager {
    * @return a (nodeSig, bitcoinSig) pair. nodeSig is the signature of the channel announcement with our node's
    *         private key, bitcoinSig is the signature of the channel announcement with our funding private key
    */
-  def signChannelAnnouncement(fundingKeyPath: DeterministicWallet.KeyPath, chainHash: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, remoteFundingKey: PublicKey, features: Features): (ByteVector64, ByteVector64)
-
-  /**
-   * Sign a digest, primarily used to prove ownership of the current node
-   *
-   * When recovering a public key from an ECDSA signature for secp256k1, there are 4 possible matching curve points
-   * that can be found. The recoveryId identifies which of these points is the correct.
-   *
-   * @param digest     SHA256 digest
-   * @param privateKey private key to sign with, default the one from the current node
-   * @return a (signature, recoveryId) pair. signature is a signature of the digest parameter generated with the
-   *         private key given in parameter. recoveryId is the corresponding recoveryId of the signature
-   */
-  def signDigest(digest: ByteVector32, privateKey: PrivateKey = nodeKey.privateKey): (ByteVector64, Int)
+  def signChannelAnnouncement(localNodeSecret: PrivateKey, fundingKeyPath: DeterministicWallet.KeyPath, chainHash: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, remoteFundingKey: PublicKey, features: Features): (ByteVector64, ByteVector64)
```

Comment:
This function becomes a bit weird now, doesn't it?
Maybe we should clearly separate signing the channel announcement with the node's key and the channel key; this will result in two distinct `signChannelAnnouncement` methods, each returning only one signature, one defined in `ChannelKeyManager` and the other in `NodeKeyManager`.

Reply:
See d7418a97
Does it seems right to you?

Reply:
Yes that's perfect.